bpf/bpf_lxc.c

```diff
@@ -569,6 +572,10 @@ static inline int handle_ipv4(struct __sk_buff *skb)
 		union macaddr host_mac = HOST_IFINDEX_MAC;
 		int ret;
 
+		// Capture the packet before the packet's destination address and port
+		// are rewritten.
+		send_trace_notify(skb, TRACE_TO_PROXY, SECLABEL, 0, 0, HOST_IFINDEX);
```

Comment:
The primary purpose of these trace messages so far was for debugging/troubleshooting of problems. In order to allow for this purpose to be fulfilled, the trace must happen right before the redirect when there is no chance of the packet being dropped anymore. With this move, that is no longer guaranteed.

There is already a trace message inside of `ipv4_redirect_to_host_port()`:
```
        cilium_dbg_capture(skb, DBG_CAPTURE_PROXY_PRE, old_port);
```

I suggest that we use/extend that instead of having two duplicate trace messages at the same spot.

Reply:
Done. I just replaced `cilium_dbg_capture` with `cilium_trace_notify` since the trace message is a superset of the dbg capture message.

Reply:
I also added a DBG_CAPTURE_PROXY_POST dbg capture post-NAT into ipv6_redirect_to_host_port for consistency with ipv4_redirect_to_host_port. I couldn't add a DBG_REV_PROXY_UPDATE dbg message since there's no space for IPv6 addresses in that message.